code/modules/buildmode/submodes/forcemove.dm

```diff
@@ -0,0 +1,55 @@
+// This tool calls forceMove() on an object/mob that is selected with right-click.
+// Once the atom is moved, its selection is cleared to avoid any further moves by accident.
+/datum/buildmode_mode/forcemove
+	key = "forcemove"
+	var/atom/selected_atom		// The atom we want to move
+
+/datum/buildmode_mode/forcemove/show_help(mob/user)
+	to_chat(user, "<span class='notice'>***********************************************************</span>")
+	to_chat(user, "<span class='notice'>Left Mouse Button on obj/mob = Select destination</span>")
+	to_chat(user, "<span class='notice'>Right Mouse Button on obj/mob = Select atom to move</span>")
+	to_chat(user, "<span class='notice'><b>Notice:</b> You need to select the movable atom first, then left-click its destination.</span>")
+	to_chat(user, "<span class='notice'>***********************************************************</span>")
+
+/datum/buildmode_mode/forcemove/handle_click(user, params, atom/A)
+	var/list/pa = params2list(params)
+	var/left_click = pa.Find("left")
+	var/right_click = pa.Find("right")
+
+	// Selecting the atom to move
+	if(right_click)
+		if(!ismovable(A) || iseffect(A))
+			to_chat(user, "<span class='danger'>You cannot move effects, turfs or areas.</span>")
+			return
+
+		selected_atom = A
+		to_chat(user, "<span class='notice'>'[selected_atom]' is selected to be moved.</span>")
+		return
+
+	// Selecting the destination to move to
+	if(!left_click)
+		return
+
+	var/atom/destination = A
+
+	if(!selected_atom)
+		to_chat(user, "<span class='danger'>Select an atom to move first (with right-click).</span>")
+		return
+
+	// Block these as they can only lead to issues
+	if(iseffect(destination) || isobserver(destination))
```

Comment:
instead of doing `istype()` checks here. Can you use `is_type_in_list()` and store a `var/static/list` of types that are banned for these datums.

This way its really easy to add more blacklisted things down the line (when an admin inevitably breaks something with this). Also here's a few more types that need to be added to the blacklist

- All types of `/obj/machinery/atmospherics` that are currently anchored <--- forcemoving these while anchored may break pipenets
- all types of `/obj/structure/disposalpipe` that are anchored, same reason for above
- all types of `/obj/machinery/power` that are anchored <---- should not be moved off of wire nodes via forcemove
- `/obj/structure/cable` <--- Shouldn't be moved off a turf or into anything via forcemove
- `/obj/machinery/field/containment` <--- should not be moved via forcemove

Either they need to go in the blacklist or they need special behavior overrides for `forcemove()` to disconnect them from station systems properly.